swap_meet/vendor.py

```diff
@@ -1,2 +1,63 @@
 class Vendor:
-    pass
+    def __init__(self, inventory=[]):
+        self.inventory=inventory
+    def add(self, item): 
+        self.inventory.append(item)
+        return item
+    def remove(self, item): 
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+    def get_by_category(self,category):
+        cat_list=[]
+        for item in self.inventory:
+            if item.category==category:
+                cat_list.append(item)
+        return cat_list
+    def swap_items(self, Vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in Vendor.inventory:
+            self.remove(my_item)
+            Vendor.add(my_item)
+            Vendor.remove(their_item)
+            self.add(their_item)
+            return True
+        else:
+            return False
+    def swap_first_item(self, Vendor):
+        
+        if self.inventory==[] or Vendor.inventory==[]:
+            return False
+        else:
+            self_first_item=self.inventory[0]
+            Vendor_first_item=Vendor.inventory[0]
+            self.remove(self_first_item)
+            Vendor.add(self_first_item)
+            Vendor.remove(Vendor_first_item)
+            self.add(Vendor_first_item)
+            return True
+    def get_best_by_category(self, category):
+        list_of_items_with_cat=self.get_by_category(category)
+        cond_list=[]
+        if not list_of_items_with_cat:
+            return None
+        else:
+            for item in list_of_items_with_cat:
+                cond_list.append(item.condition)
+            for item in list_of_items_with_cat:
+                if item.condition==max(cond_list):
+                    return item
```

Comment:
Even though this is a good way to approach it, here is a way to do it using one less for loop. You can initialize two variables and then do a comparison: 
```suggestion
      highest = 0
      best_item = None

       for item in self.inventory:
            if item.category == category and item.condition > highest:
                best_item = item
                highest = item.condition

       return best_item
```